src/easylink/step.py

```diff
@@ -388,17 +402,35 @@ def _create_looped_graph(self, num_loops: int) -> nx.MultiDiGraph:
         return graph
 
     def _get_loop_slot_mappings(self, num_loops: int) -> nx.MultiDiGraph:
-        input_mappings = [
-            SlotMapping("input", self.name, slot, f"{self.name}_loop_1", slot)
-            for slot in self.input_slots
-        ]
+        """Get the appropriate slot mappings for the CompositeStep based on the number of loops
+        and the non-self-edge input and output slots."""
+        input_mappings = []
+        self_edge_input_slots = {edge.input_slot for edge in self.self_edges}
+        external_input_slots = self.input_slots.keys() - self_edge_input_slots
+        for input_slot in self_edge_input_slots:
+            input_mappings.append(
+                SlotMapping("input", self.name, input_slot, f"{self.name}_loop_1", input_slot)
+            )
+        for input_slot in external_input_slots:
+            input_mappings.extend(
+                [
+                    SlotMapping(
+                        "input", self.name, input_slot, f"{self.name}_loop_{n+1}", input_slot
```

Comment:
should this be `..._loop_{n+2}` since loop 1 is already taken up by the self-edge input slots?

Reply:
I think this is correct, but would be clearer if `SlotMapping` was instantiated with kwargs.